hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -123,20 +138,14 @@ def execute_command(command, do_say, do_log)
     begin
       stdout_stderr, status = Open3.capture2e(*Kafo::PuppetCommand.format_command(command))
     rescue Errno::ENOENT
-      log_and_say(:error, "Command #{command} not found", do_say, do_log)
-      return false
+      return [nil, false]
```

Comment:
Nit: I think `nil` is better than `false`. From type consistency it then becomes either an object or nil which feels more natural to me.

Reply:
Sorry, I do not understand. The second argument is the status, and the command failed to execute. So the status of it having executed is false. That feels like type consistency.

Reply:
Currently the return type is (in yard style) `Array<(nil, Boolean)>` (command not found) or `Array<(String, Process::Status)>` (command ran). I'm arguing that `Array<(nil, nil)>` makes more sense to me in the former case because it allows cleaner code later on:
```ruby
false&.success? # raises NoMethodError: undefined method `success?' for false:FalseClass
nil&.success? # returns nil
```

It's a nit since this isn't actually used, but since you're creating an API this is something I think about.

An alternative is to return `nil` instead of `[nil, nil]` and capture it that way but I'm not sure if that's a much cleaner API.

Reply:
Ahh yea, thats a bug in the overall methods usage on my part and inconsistency. Now `execute_command` returns `Array(nil|string, Boolean)` consistently and is used as such by the `execute` and `execute!` methods which fixes a possible bug in those hooks using `execute` as well.

Reply:
That's another way that I'm also fine with. All methods except `success?` are irrelevant anyway for the consumer of `execute_command` since the process has already completed, except perhaps `exitstatus`. Not sure if we may want that in the future.